chips/stm32f4xx/src/clocks/clocks.rs

```diff
@@ -0,0 +1,926 @@
+// Licensed under the Apache License, Version 2.0 or the MIT License.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+// Copyright OxidOS Automotive SRL.
+//
+// Author: Ioan-Cristian CÎRSTEA <ioan.cirstea@oxidos.io>
+
+#![deny(dead_code)]
+#![deny(missing_docs)]
+#![deny(unused_imports)]
```

Comment:
Do we need the dead_code and unused_imports? Aren't those already on?

Reply:
For me, they are not on by default. Any missing documentation and unused imported are checked during CI. However, I prefer that compilation fails before I can even make `git push`. Any new future developer can simply switch the `#![deny]` to `#![warning]` if they found it annoying or trying to develop something quickly.